c++/src/capnp/compat/byte-stream-test.c++

```diff
@@ -64,6 +70,38 @@ KJ_TEST("KJ -> ByteStream -> KJ without shortening") {
     promise.wait(waitScope);
   }
 
+  {
+    // Write more than 1 << 16 bytes at once to exercise write splitting.
+    auto str = makeString(1 << 17);
+    auto promise = wrapped->write(str.begin(), str.size());
+    KJ_EXPECT(!promise.poll(waitScope));
+    expectRead(*pipe.in, str).wait(waitScope);
+    promise.wait(waitScope);
+  }
+
+  {
+    // Write more than 1 << 16 bytes via an array to exercise write splitting.
+    auto str = makeString(1 << 18);
+    auto pieces = kj::heapArrayBuilder<kj::ArrayPtr<const kj::byte>>(4);
+
+    // Two 2^15 pieces will be combined.
+    pieces.add(kj::ArrayPtr(reinterpret_cast<kj::byte*>(str.begin()), 1 << 15));
+    pieces.add(kj::ArrayPtr(reinterpret_cast<kj::byte*>(str.begin() + (1 << 15)), 1 << 15));
+
+    // One 2^16 piece will be written alone.
+    pieces.add(kj::ArrayPtr(reinterpret_cast<kj::byte*>(
+        str.begin() + (1 << 16)), 1 << 16));
+
+    // One 2^17 piece will be split.
+    pieces.add(kj::ArrayPtr(reinterpret_cast<kj::byte*>(
+        str.begin() + (1 << 16)), str.size() - (1 << 17)));
+
+    auto promise = wrapped->write(pieces);
+    KJ_EXPECT(!promise.poll(waitScope));
+    expectRead(*pipe.in, str).wait(waitScope);
+    promise.wait(waitScope);
```

Comment:
With my latest change this wait results in

```
kj/async.c++:1593: failed: Nothing to wait for; this thread would hang forever.
```

I've read and re-read the code. I put in prints and it the splitting follows the path I expect the correct number of times. Maybe someone can spot what I'm missing? I'm not feeling well today, I'm sure it's something dumb.